autogen/main/cluster.tf.tmpl

```diff
@@ -882,6 +882,14 @@ resource "google_container_node_pool" "windows_pools" {
             gpu_driver_version = lookup(each.value, "gpu_driver_version", "")
           }
         }
+
+        dynamic "gpu_sharing_config" {
+          for_each = lookup(each.value, "gpu_sharing_strategy", "") != "" ? [1] : []
+          content {
+            gpu_sharing_strategy = lookup(each.value, "gpu_sharing_strategy", "")
+            max_shared_clients_per_gpu = lookup(each.value, "max_shared_clients_per_gpu", 2)
```

Comment:
do we want to default to `2` shares?
`gcloud` has no default: https://cloud.google.com/sdk/gcloud/reference/container/node-pools/create

Reply:
It seems like 2 is the default if you omit the `cloud.google.com/gke-max-shared-clients-per-gpu` https://cloud.google.com/kubernetes-engine/docs/concepts/timesharing-gpus#gpu_time-sharing_or_nvidia_mps Hence the choice of 2 here as well.